### ISSUE TITLE
CommandLoader: Decouple files from implementation

### ISSUE THREAD

**Issue by ybakos** [osu-cascades/hackbot#37]
The constructor is explicitly dependent on the files in the particular directory. This shouldn't be explicit in the body of the constructor.

The CommandLoader `constructor` should have one parameter: the list of files.
And it should not have a `getFiles` property, (which is a poor name too).

/cc @Otis0620 @ctsstc

**Comment by ctsstc** [comment]
Could rollback to here, implement the parameter, update tests.

https://github.com/osu-cascades/hackbot/blob/d7d89eee4dd8227e75b984f80eb337fde702515b/library/command-loader.js

**Comment by ybakos** [comment]
Nah, that old commit doesn't help much. When refactoring, start from where you are at. You will find that resolving this issue will reveal further refactorings.

**Comment by Otis0620** [comment]
@Yong Is there anything about this version that you would keep https://github.com/osu-cascades/hackbot/blob/d7d89eee4dd8227e75b984f80eb337fde702515b/library/command-loader.js

**Comment by ybakos** [comment]
@Otis0620 I wouldn't worry about that. Look at the current implementation, think on that, and go from there. 😄

**Comment by ybakos** [comment]
To illustrate:

```
files = glob...
commandLoader = new CommandLoader(files); // no major work in constructor!
commands = commandLoader.getCommandClassList(); // return array of command classes
```

Just an example. I am not suggesting that explicit API.

**Comment by Otis0620** [comment]
@yong. So the work that loadCommandClasses method is doing is unnecessary?

**Comment by ybakos** [comment]
A dict is fine. That's not the point. If you resolve this issue (look at the title and top post) the rest should reveal itself.

**Comment by ctsstc** [comment]
Should be fixed by #46 🤞

**Comment by ybakos** [comment]
Thanks @ctsstc
